payu/models/cesm_cmeps.py

```diff
@@ -190,6 +185,72 @@ def setup(self):
                 # TODO: copied this from other models. Surely we want to exit here or something
                 print('payu: error: Unable to find mod_def.ww3 file in input directory')
 
+    def _setup_checks(self):
+        # check pelayout fits within requested cpucount
+        cpucount = int(
+            self.expt.config.get('ncpus', multiprocessing.cpu_count())
+        )
+        all_realms = self.realms
+        for realm in all_realms:
+            ntasks = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_ntasks"))
+            nthreads = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_nthreads"))
+            rootpe = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_rootpe"))
+            pestride = int(self.runconfig.get("PELAYOUT_attributes", f"{realm}_pestride"))
+
```

Comment:
Does this improvement work better? In case nthreads is less than 1. 
```suggestion

            if nthreads < 1:
                raise ValueError(f"The number of {realm}_nthreads ({nthreads}) in "
                                 f"{NUOPC_CONFIG} must be at least 1.")
                                 
```

Reply:
It depends if we want to support setting it automatically (i.e. `-99`) ?